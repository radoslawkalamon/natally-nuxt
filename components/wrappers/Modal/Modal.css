.modal {
  background-color: var(--color-background-modal);
  box-sizing: border-box;
  height: 100%;
  left: 0;
  overflow-y: auto;
  padding: var(--gap);
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 9;
}

.modal__frame {
  background-color: var(--color-background-1);
  border-radius: var(--border-radius);
  box-shadow: var(--box-shadow-modal);
  box-sizing: border-box;
  margin: 0 auto;
  max-width: var(--max-width-body);
  padding: var(--padding-modal);
  width: 100%;
}

.modal__canvas {
  display: flex;
  flex-flow: column nowrap;
  gap: var(--gap);
}

.modal__canvas >>> > *:first-child {
  margin-top: 0;
}

.modal__canvas >>> > *:last-child {
  margin-bottom: 0;
}

.modal__title {
  margin-bottom: var(--gap-section);
  text-align: center;
}

.fade-enter-active,
.fade-leave-active {
  transition: opacity var(--time-animation);
}

.fade-enter,
.fade-leave-to {
  opacity: 0;
}
