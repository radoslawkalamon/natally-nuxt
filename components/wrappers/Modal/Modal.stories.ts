import type { Meta, Story } from '@storybook/vue'
import WrappersModal from '@/components/wrappers/Modal/Modal.vue'

const meta: Meta = {
  title: 'Wrappers / Modal',
  component: WrappersModal,
  argTypes: {
    default: {
      description: '-',
      table: {
        type: null,
        defaultValue: null
      }
    }
  }
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: {
    WrappersModal
  },
  props: Object.keys(argTypes),
  template: `<WrappersModal v-bind="$props">
    Pellentesque ullamcorper elit a nisl accumsan, eu scelerisque justo rutrum. Curabitur ullamcorper ex non lacus sodales, eu placerat purus bibendum. Integer sem quam, pharetra at sem nec, volutpat aliquam tellus. Phasellus in leo lacus. Pellentesque justo nisl, dictum nec fringilla quis, eleifend vitae neque. Proin a augue et massa bibendum hendrerit. Suspendisse ut turpis fermentum, ornare nisl vitae, luctus tellus. Phasellus vel libero elit. Nulla eget est eu neque pretium vestibulum. Morbi hendrerit venenatis augue eget suscipit. Duis ex enim, lacinia sit amet dignissim sed, mattis vitae leo. Nunc aliquam rutrum leo, vitae molestie nisl sodales ut. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae
  </WrappersModal>`
})

export const Default: Story = Template.bind({})
Default.args = {
  title: 'Polityka prywatno≈õci'
}
