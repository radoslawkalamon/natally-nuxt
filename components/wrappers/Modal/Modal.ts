import Vue from 'vue'

export default Vue.extend({
  name: 'WrappersModal',
  components: {
    ComponentsTitle: () => import(/* webpackChunkName: "components" */'@/components/components/Title/Title.vue')
  },
  props: {
    title: {
      type: String,
      required: true
    }
  }
})
