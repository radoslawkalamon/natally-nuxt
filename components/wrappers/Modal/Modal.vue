<template>
  <transition name="fade">
    <article class="modal">
      <div class="modal__frame">
        <ComponentsTitle
          class="modal__title"
          :type="1"
          :design="2"
          :title="title"
        />
        <div class="modal__canvas">
          <slot />
        </div>
      </div>
    </article>
  </transition>
</template>

<script src="./Modal.ts" lang="ts"></script>
<style src="./Modal.css" scoped></style>
