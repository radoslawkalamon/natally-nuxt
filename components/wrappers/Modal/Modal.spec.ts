import Modal from './Modal.vue'
import { shallRender, shallHaveSlot } from '@/devtools/jest.shared.spec'

describe('Wrappers / Modal', () => {
  const defaultOptions = {
    propsData: {
      title: 'Test title'
    },
    stubs: {
      ComponentsTitle: {
        template: '<h1 data-stub="components-title" />'
      }
    }
  }

  shallRender(Modal, defaultOptions)
  shallHaveSlot(Modal, 'default', defaultOptions)
})
