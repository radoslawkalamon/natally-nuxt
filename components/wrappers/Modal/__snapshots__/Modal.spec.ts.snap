// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrappers / Modal Integration shall pass integration sanity test 1`] = `
"<transition-stub name="fade">
  <article class="modal">
    <div class="modal__frame">
      <h1 class="modal__title title title--design-2 title--underscore">Test title</h1>
      <div class="modal__canvas">This is default slot test</div>
    </div>
  </article>
</transition-stub>"
`;

exports[`Wrappers / Modal Unit shall pass unit sanity test 1`] = `
"<transition-stub name="fade">
  <article class="modal">
    <div class="modal__frame">
      <componentstitle-stub type="1" design="2" title="Test title" class="modal__title"></componentstitle-stub>
      <div class="modal__canvas">This is default slot test</div>
    </div>
  </article>
</transition-stub>"
`;
